### DISCUSSION TITLE
Finding FIELD directives on a SelectedField

### DISCUSSION THREAD

**Discussion by ee-usgs** [graphql-java/graphql-java#3253] [Q&A]
I've created a `@tabular(name: String!) on FIELD` directive.  It's only defined in my schema - there are no associated classes.  The purpose of the directive is to provide the column name for a field if it is later exported as tabular data (CSV, TSV, etc.).

For a query like this:

```GraphQL
{
  resultConnection(first: 100) {
    pageInfo {
      hasNextPage, endCursor, tabularMappings { path, name }
    }
    nodes {
      FIRST_FIELD,
      SOME_FIELD_I_DONT_LIKE_THE_NAME_OF @tabular(name: "A_BETTER_NAME")
    }
  }
}
```

I have one DataFetcher that is wired to provide the entire response.  The intention is that it fills in the `tabularMappings` section with the paths and names to use for exporting the values to tabular formats, e.g. like this:

```json
        "tabularMappings": [
          {
            "path": "FIRST_FIELD",
            "name": "firstField" // pretty column name
          }, {
            "path": "SOME_FIELD_I_DONT_LIKE_THE_NAME_OF",
            "name": "A_BETTER_NAME" // The user is specifically requesting this name
          }
        ]
```

The problem is,  cannot find the directive anywhere within the `SelectedField` or anywhere else in the DataFetchingEnvironment.  Where can I find it?

Any suggestions appreciated,

--Eric Everman

**Comment by bbakerman** [comment]
`graphql.schema.DataFetchingEnvironment#getQueryDirectives` will give you access to the query directives on a field

`SOME_FIELD_I_DONT_LIKE_THE_NAME_OF @tabular(name: "A_BETTER_NAME")`

I am not sure this will work the way you think it will.  The engine REALLY wants to return data results using the field names as they are defined in the schema

A client can use a field alias to rename field names in results

```
nodes {
      FIRST_FIELD,
     A_BETTER_NAME :  SOME_FIELD_I_DONT_LIKE_THE_NAME_OF
    }
```

https://graphql.org/learn/queries/#aliases

**Reply by eeverman** [reply]
I will try `graphql.schema.DataFetchingEnvironment#getQueryDirectives`.

Just to be clear, the intention is not to create an alias: It is to create some extra metadata, returned in the pageInfo section, that can be used by a downstream system to export the data into a tabular format.  In some cases the tabular export name is not a valid GraphQL field name.
